### PR TITLE
Creates dirs per db and populates with files

### PR DESCRIPTION
**Pull request by RiggsRiggs**
The object of the program is to:

Take a list of property references from a database
Make a directory for each property reference
Take a list of files
Where filename starts with property reference, move it to its respective folder
Zip the folders

The DBWork branch includes:
Addition of data access layer
Creation of directories based on db content, rather than first part of filenames

### CODE REVIEW

**Review by steve-shenton-ea — APPROVED** [review]

File: DocAggregatorDotnetCore/TomAggregator.cs [hunk 1]
```diff
@@ -38,15 +33,22 @@ public void DoSomeWork()
 
             Directory.CreateDirectory(outputDirectoryPath);
 
-            //Console.WriteLine("New folders to go in: " + outputDirectoryPath);
+            //make array of property refs
+            var ctx = new RiggsTestContext();
```
Comment (benscobie) [inline DocAggregatorDotnetCore/TomAggregator.cs]: You should ideally use a `using` statement when creating an Entity Framework context, like so:

using(var ctx = new RiggsTestContext()) {
  // CODE HERE
}

This ensures all the contexts resources are cleaned up straight away, and it's nice to consider that single block of DB work as a transaction.

File: DocAggregatorDotnetCore/TomAggregator.cs [hunk 2]
```diff
@@ -38,15 +33,22 @@ public void DoSomeWork()
...
+            var listOfDocs = ctx.PropDocs.Select(p => p).ToList();
+            var docsArray = listOfDocs.ToArray();
```
Comment (benscobie) [inline DocAggregatorDotnetCore/TomAggregator.cs]: I believe you can remove this line as you can foreach over your listOfDocs as it's a list.

`foreach` works on anything that implements the IEnumerable interface, which List does.

File: DocAggregatorDotnetCore/TomAggregator.cs [hunk 3]
```diff
@@ -38,15 +33,22 @@ public void DoSomeWork()
 
             Directory.CreateDirectory(outputDirectoryPath);
 
-            //Console.WriteLine("New folders to go in: " + outputDirectoryPath);
+            //make array of property refs
+            var ctx = new RiggsTestContext();
+            var listOfDocs = ctx.PropDocs.Select(p => p).ToList();
```
Comment (benscobie) [inline DocAggregatorDotnetCore/TomAggregator.cs]: You don't need the Select here as the default is to bring everything back.

Comment (benscobie) [reply DocAggregatorDotnetCore/TomAggregator.cs]: I'm not sure if you've read about Change Tracking in Entity Framework yet, but if you aren't planning to update any of the results coming back from this call, you could use .AsNoTracking() to opt out the Change Tracking behaviour, which will reduce memory footprint etc.

File: DocAggregatorDotnetCore/TomAggregator.cs [hunk 4]
```diff
@@ -38,15 +33,22 @@ public void DoSomeWork()
...
+
 
-            foreach (var file in fileInfos)
+            //new make directories
+            foreach (var item in docsArray)
             {
+                Console.WriteLine(item.PropRef);
                 //Console.WriteLine(file.Name.Substring(0, offset));
-                var propertyDirectory = Path.Combine(outputDirectoryPath, file.Name.Substring(0, offset));
+                var propertyDirectory = Path.Combine(outputDirectoryPath,item.PropRef.ToString());
```
Comment (benscobie) [inline DocAggregatorDotnetCore/TomAggregator.cs]: I can see that PropRef is a nullable integer. You should ideally include a null check before using it, or perhaps adopt the listOfDocs query so it does not bring back these if you don't plan on doing anything with those that are null.

**Review by benscobie — CHANGES_REQUESTED** [review]
A couple of tweaks possible.

File: DocAggregatorDotnetCore/TomAggregator.cs (hunk 3 above)
Comment (RiggsRiggs) [reply DocAggregatorDotnetCore/TomAggregator.cs]: Fixed!

File: DocAggregatorDotnetCore/TomAggregator.cs (hunk 4 above)
Comment (RiggsRiggs) [reply DocAggregatorDotnetCore/TomAggregator.cs]: Fixed!

**Review by benscobie — APPROVED** [review]

**Pull request merged**